src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,215 @@
+        const startPositionX = this.renderEngine.parent.positionX;
+
+        this.interactionsEngine.setCursor('grabbing');
+
+        this.renderEngine.tryToChangePosition(position.deltaX);
+
+        if (startPositionX !== this.renderEngine.parent.positionX) {
+            this.renderEngine.parent.render();
+        }
+    }
+
+    handleMouseUp() {
+        this.interactionsEngine.clearCursor();
+    }
+
+    setData(data: TimeseriesPoint[]) {
+        this.data = data;
+
+        let min = Number.MAX_VALUE;
+        let max = Number.MIN_VALUE;
+        let first = Number.MAX_VALUE;
+        let last = Number.MIN_VALUE;
+
+        this.data.forEach(([timestamp, value]) => {
+            if (value < min) {
+                min = value;
+            }
+            if (value > max) {
+                max = value;
+            }
+
+            if (timestamp < first) {
+                first = timestamp;
+            }
+            if (timestamp > last) {
+                last = timestamp;
+            }
+        });
+
+        this.summary = {
+            min,
+            max,
+            first,
+            last,
+        };
+    }
+
+    calcRect(start: number, duration: number, isEnd: boolean) {
+        const w = duration * this.renderEngine.zoom;
+
+        return {
+            x: this.renderEngine.timeToPosition(start),
+            w: isEnd ? (w <= 0.1 ? 0.1 : w >= 3 ? w - 1 : w - w / 3) : w,
+        };
+    }
+
+    findClosestDataPoint(timestamp: number): TimeseriesPoint {
+        return [timestamp, -1];
+    }
+
+    override renderTooltip() {
+        const mouse = this.interactionsEngine.getGlobalMouse();
+        //QUESTION: how do i convert mouse.x to position
+        const timestamp = mouse.x;
+        const tp = this.findClosestDataPoint(timestamp);
+        //TODO - ...
+
+        this.renderEngine.renderTooltipFromData(
+            [
+                { text: `Value:${tp[1]}` },
+                {
+                    text: `Timestamp: ${tp[0]}ms`,
+                },
+            ],
+            this.interactionsEngine.getGlobalMouse()
+        );
+        return true;
+    }
+
+    private normalizeValue(value: number, heightPerValueUnit: number) {
+        return this.height - value * heightPerValueUnit;
+    }
+
+    override render() {
+        if (this.data?.length === 0 || !this.summary) {
+            return;
+        }
+
+        const timestampStart = this.renderEngine.positionX;
+        const width = this.renderEngine.getRealView();
+        const timestampEnd = this.renderEngine.positionX + width;
+
+        const positionStart = this.renderEngine.timeToPosition(timestampStart);
+        const positionEnd = this.renderEngine.timeToPosition(timestampEnd);
+
+        const heightPerValueUnit =
+            (this.height - this.styles.padding) / Math.max(this.summary.max - this.summary.min, 1);
+
+        this.interactionsEngine.addHitRegion(RegionTypes.CLUSTER, 'not-used', 0, 0, width, this.height);
```

Comment:
This seemed to help - but 100% sure if it does or not

Reply:
Yes, it works. But you need to add a new enum for the plugin.)